File: yeonkyung/이진 변환 반복하기.py
def binary(n):
    if n == 0:
        return ""
    return binary(n//2) + str(n%2)

def solution(s):
    answer = [0, 0]

    while True:
        if s == "1":
            break

        count = 0
        for e in s:
            if e == "0":
                count += 1
        answer[1] += count
        length = len(s) - count
        s = binary(length)
        answer[0] += 1

    return answer

if __name__ == "__main__":
    s = "1111111"
    print(solution(s))

File: Doodream/이진 변환 반복하기.js
function solution(s) {
    var reCount = 0;
    var zeroCount = 0;
    while (s.length > 1) {
        var beforeLength = s.length;
        s = s.split('0').join('');
        var afterLength = s.length;
        zeroCount += beforeLength - afterLength;
        s = afterLength.toString(2);
        reCount++;
    }
    var answer = [reCount, zeroCount];
    return answer;
}

var s = "110010101001";
console.log(solution(s));

